src/main/scala/com/nec/arrow/ArrowInterfaces.scala

```diff
@@ -43,65 +40,58 @@ object ArrowInterfaces {
 
   def c_double_vector(float8Vector: Float8Vector): non_null_double_vector = {
     val vc = new non_null_double_vector()
-    vc.data = float8Vector.getDataBuffer.nioBuffer().asInstanceOf[DirectBuffer].address()
+    vc.data = float8Vector.getDataBufferAddress()
 
     vc.count = float8Vector.getValueCount
     vc
   }
 
   def c_nullable_double_vector(float8Vector: Float8Vector): nullable_double_vector = {
     val vc = new nullable_double_vector()
-    vc.data = float8Vector.getDataBuffer.nioBuffer().asInstanceOf[DirectBuffer].address()
+    vc.data = float8Vector.getDataBufferAddress()
     vc.validityBuffer = float8Vector.getValidityBufferAddress
     vc.count = float8Vector.getValueCount
     vc
   }
 
   def c_nullable_varchar_vector(varCharVector: VarCharVector): nullable_varchar_vector = {
     val vc = new nullable_varchar_vector()
-    vc.data = varCharVector.getDataBuffer.nioBuffer().asInstanceOf[DirectBuffer].address()
-    vc.offsets = varCharVector.getOffsetBuffer.nioBuffer().asInstanceOf[DirectBuffer].address()
+    vc.data = varCharVector.getDataBufferAddress()
+    vc.offsets = varCharVector.getOffsetBufferAddress()
     vc.validityBuffer =
-      varCharVector.getValidityBuffer.nioBuffer().asInstanceOf[DirectBuffer].address()
+      varCharVector.getValidityBufferAddress()
     vc.count = varCharVector.getValueCount
     vc.dataSize = varCharVector.sizeOfValueBuffer()
     vc
   }
 
   def c_bounded_string(string: String): non_null_c_bounded_string = {
     val vc = new non_null_c_bounded_string()
-    vc.data =  (new BytePointer(string.length)).asBuffer
-      .put(string.getBytes())
-      .asInstanceOf[DirectBuffer]
+    vc.data =  (new BytePointer(string.length))
+      .put(string.getBytes(): _*)
```

Comment:
Are we sure this doesn't need to be null-terminated?

Reply:
Right, it stores the length separately, so I guess this is alright, but I don't even see where the data gets used!